components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -850,6 +861,49 @@ public static boolean isValidWSAPI(APIDTO api) {
         return isValid;
     }
 
+    /**
+     * Validate sandbox and production endpoint URLs.
+     *
+     * @param apiDto API DTO of the API
+     * @return validity of URLs found within the endpoint configurations of the DTO
+     */
+    public static boolean validateEndpoints(APIDTO apiDto) {
+
+        ArrayList<String> endpoints = new ArrayList<>();
+        UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_ALL_SCHEMES +
+                UrlValidator.ALLOW_LOCAL_URLS);
+
+        if (apiDto.getEndpointConfig() != null) {
+            Map endpointConfig = (Map) apiDto.getEndpointConfig();
+            endpoints.add(String.valueOf(((Map) endpointConfig
+                    .get(APIConstants.API_DATA_SANDBOX_ENDPOINTS)).get(APIConstants.API_DATA_URL)));
+            endpoints.add(String.valueOf(((Map) endpointConfig
+                    .get(APIConstants.API_DATA_PRODUCTION_ENDPOINTS)).get(APIConstants.API_DATA_URL)));
+
+            for (String endpoint : endpoints) {
+                // If url is a JMS connection url, validation is skipped. If not, validity is checked.
+                if (!endpoint.startsWith("jms:") && !urlValidator.isValid(endpoint)) {
+                    try {
+                        // If the url is not identified as valid from the above check,
+                        // next step is determine the validity of the encoded url (done through the URI constructor).
+                        URL endpointUrl = new URL(endpoint);
+                        URI endpointUri = new URI(endpointUrl.getProtocol(), endpointUrl.getAuthority(),
+                                endpointUrl.getPath(), endpointUrl.getQuery(), null);
+
+                        if (!urlValidator.isValid(endpointUri.toString())) {
+                            log.error("Invalid endpoint url " + endpointUrl);
+                            return false;
+                        }
+                    } catch (URISyntaxException | MalformedURLException e) {
+                        log.error("Error while parsing the endpoint url " + endpoint);
+                        return false;
+                    }
+                }
```

Comment:
This part of the logic can be moved to APIUtil.java static method and call that from the existing validate function here.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10715/commits/d2955ccba1428a37539279ca49f5dce722f64b1e